src/components/authors/manageAuthorPage.js

```diff
@@ -1,46 +1,37 @@
 "use strict";
 
 var React = require('react');
-var Router = require('react-router');
+var createReactClass = require('create-react-class');
+var ReactRouter = require('react-router-dom');
 var AuthorForm = require('./authorForm');
 var AuthorActions = require('../../actions/authorActions');
 var AuthorStore = require('../../stores/authorStore');
 var toastr = require('toastr');
 
-var ManageAuthorPage = React.createClass({
-	mixins: [
-		Router.Navigation
-	],
-
-	statics: {
-		willTransitionFrom: function(transition, component) {
-			if (component.state.dirty && !confirm('Leave without saving?')) {
-				transition.abort();
-			}
-		}
-	},

```

Comment:
Eliminated this code by using React Router 4's `Redirect` and `Prompt` components below instead.